File: aqa_js/automation_playwright/pom/adduserpage.js
const {Page, expect} = require('@playwright/test');
const BasePage = require('./basepage');

class AddUserPage extends BasePage {
    /**
     * @param {Page} page - Playwright Page object
     */
    constructor(page) {
        super(page);
        this.PageUrl = `${this.config.BASE_URL}/addUser`; // Replace with the actual URL of the signup page
        this.ids = {
            firstNameFieldId: '#firstName',
            lastNameFieldId: '#lastName',
            emailFieldId: '#email',
            passwordFieldId: '#password',
            submitButtonId: '#submit',
            cancelButtonId: '#cancel',
            errorMessageId: '#error'
        };
    }

    async navigate() {
        await this.page.goto(this.PageUrl);
    }

    async enter_first_name(firstNameText) {
        await this.page.fill(this.ids.firstNameFieldId, firstNameText);
    }

    async enter_last_name(lastNameText) {
        await this.page.fill(this.ids.lastNameFieldId, lastNameText);
    }

    async enter_email(emailText) {
        await this.page.fill(this.ids.emailFieldId, emailText);
    }

    async enter_password(passwordText) {
        await this.page.fill(this.ids.passwordFieldId, passwordText);
    }

    async click_submit() {
        await this.page.click(this.ids.submitButtonId);
    }

    async click_cancel() {
        await this.page.click(this.ids.cancelButtonId);
    }
    async get_error_message() {
        // Wait for the error message to be visible
        await this.page.waitForSelector(this.ids.errorMessageId, { state: 'visible' }); 
        return await this.page.textContent(this.ids.errorMessageId);
    }
}

module.exports = AddUserPage;

File: aqa_python/automation_playwright/pom/adduser_page.py
# pages/add_user_page.py

from pom.base_page import BasePage

class AddUserPage(BasePage):
    def __init__(self, page):
        super().__init__(page)
        self.page_url = f"{self.config.BASE_URL}/addUser"  # Update this URL as needed
        self.selectors = {
            "first_name_field": "#firstName",
            "last_name_field": "#lastName",
            "email_field": "#email",
            "password_field": "#password",
            "submit_button": "#submit",
            "cancel_button": "#cancel",
            "error_message": "#error"
        }

    def navigate(self):
        self.page.goto(self.page_url)
        self.page.wait_for_timeout(200)

    def enter_first_name(self, first_name: str):
        self.page.fill(self.selectors["first_name_field"], first_name)
        self.page.wait_for_timeout(200)

    def enter_last_name(self, last_name: str):
        self.page.fill(self.selectors["last_name_field"], last_name)
        self.page.wait_for_timeout(200)

    def enter_email(self, email: str):
        self.page.fill(self.selectors["email_field"], email)
        self.page.wait_for_timeout(200)

    def enter_password(self, password: str):
        self.page.fill(self.selectors["password_field"], password)
        self.page.wait_for_timeout(200)

    def click_submit(self):
        self.page.click(self.selectors["submit_button"])
        self.page.wait_for_timeout(200)

    def click_cancel(self):
        self.page.click(self.selectors["cancel_button"])
        self.page.wait_for_timeout(200)

    def get_error_message(self) -> str:
        self.page.wait_for_selector(self.selectors["error_message"], state="visible")
        return self.page.text_content(self.selectors["error_message"])
